### DISCUSSION TITLE
Fetching refferences queries in one go?

### DISCUSSION THREAD

**Discussion by Arthurdw** [prisma/prisma#5332] [Q&A] [answered]
If I have the following ERD:
```
TABLE: FOO
    id    int
    value string
    bar   int refferences BAR.id

TABLE: BAR
    id    int
    value string
``` 
How would I fetch this  as the following object by foo its Id? 
```ts
{
  id: 1,
  value: "Foo",
  bar: {
    id: 2,
    value: "bar",
  },
}
``` 
(In one go)
Because the current method would be: 
```ts
const foo = await prisma.FOO.findFirst({
  where: { id: 1},
});

const bar = await prisma.BAR.findFirst({
  where: { id: foo.bar },
});

const data = {
  id: foo.id,
  value: foo.value,
  bar: {
    id: bar.id,
    value: bar.value
  },
};
```

**Comment by ryands17** [comment, marked as answer]
@Arthurdw 👋 

Assuming your schema is as follows:

```prisma
model Foo {
  id    Int    @id @default(autoincrement())
  value String
  bar   Bar    @relation(fields: [barId], references: [id])
  barId Int
}

model Bar {
  id    Int    @id @default(autoincrement())
  value String
  Foo   Foo?
}
```

You can directly use `include` to fetch the relation model as follows:

```ts
const foo = await prisma.foo.findFirst({
    where: { id: 1 },
    include: { bar: true },
})
```

I would suggest reading [this](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries) part of the docs to have a better idea about relation queries.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
